docs/book/src/user/quick-start.md

```diff
@@ -0,0 +1,477 @@
+  diskGiB: ${VSPHERE_DISK_GIB}
+  template: "${VSPHERE_TEMPLATE}"
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: capi-quickstart-controlplane-0
+spec:
+  initConfiguration:
+    nodeRegistration:
+      name: "{{ ds.meta_data.hostname }}"
+      criSocket: "/var/run/containerd/containerd.sock"
+      kubeletExtraArgs:
+        cloud-provider: vsphere
+  clusterConfiguration:
+    apiServer:
+      extraArgs:
+        cloud-provider: vsphere
+        cloud-config: /etc/kubernetes/vsphere.conf
+      extraVolumes:
+      - name: "cloud-config"
+        hostPath: /etc/kubernetes/vsphere.conf
+        mountPath: /etc/kubernetes/vsphere.conf
+        readOnly: true
+        pathType: File
+    controllerManager:
+      extraArgs:
+        cloud-provider: vsphere
+        cloud-config: /etc/kubernetes/vsphere.conf
+      extraVolumes:
+      - name: "cloud-config"
+        hostPath: /etc/kubernetes/vsphere.conf
+        mountPath: /etc/kubernetes/vsphere.conf
+        readOnly: true
+        pathType: File
+  files:
+  - path: /etc/kubernetes/vsphere.conf
+    owner: root:root
+    permissions: "0600"
+    encoding: base64
+    content: |
+      ${CLOUD_CONFIG_B64ENCODED}
+  users:
+  - name: capv
+    sudo: "ALL=(ALL) NOPASSWD:ALL"
+    sshAuthorizedKeys:
+    - "${SSH_AUTHORIZED_KEY}"
+  preKubeadmCommands:
+  - hostname "{{ ds.meta_data.hostname }}"
+  - echo "::1         ipv6-localhost ipv6-loopback" >/etc/hosts
+  - echo "127.0.0.1   localhost {{ ds.meta_data.hostname }}" >>/etc/hosts
+  - echo "{{ ds.meta_data.hostname }}" >/etc/hostname
+```
+{{#/tab }}
+{{#/tabs }}
+
+After the controlplane is up and running, we should apply the addons to install Calico.
+
+{{#tabs name:"tab-usage-addons" tabs:"AWS,vSphere"}}
+{{#tab AWS}}
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-aws/v0.4.1/examples/addons.yaml
+```
+{{#/tab }}
+{{#tab vSphere}}
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-vsphere/v0.5.0/examples/addons.yaml
+```
+
+{{#/tab }}
+{{#/tabs }}
+
+At this point, our control plane should be up and in `Ready` state, check the status using `kubectl get nodes`.
+
+Finishing up, we'll create a single node _MachineDeployment_.
+
+{{#tabs name:"tab-usage-machinedeployment" tabs:"AWS,vSphere"}}
+{{#tab AWS}}
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha2
+kind: MachineDeployment
+metadata:
+  name: capi-quickstart-md-0
+  labels:
+    cluster.x-k8s.io/cluster-name: capi-quickstart
+    nodepool: nodepool-0
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: capi-quickstart
+      nodepool: nodepool-0
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: capi-quickstart
+        nodepool: nodepool-0
+    spec:
+      version: v1.15.3
+      bootstrap:
+        configRef:
+          name: capi-quickstart-md-0
```

Comment:
I'd use a more generic name, otherwise it'll lead to odd names for the generated config resources.